recipes/pyre/build.sh

```diff
@@ -0,0 +1,31 @@
+#!/bin/sh
+set -euo pipefail
+
+# Is this macOS?
+if [ -v MACOSX_DEPLOYMENT_TARGET ]; then
```

Comment:
What's the proper way to detect when building for macOS? Tried to base this on conda's [macOS SDK](https://docs.conda.io/projects/conda-build/en/latest/resources/compiler-tools.html#macos-sdk) instructions but this doesn't seem to be triggered in the osx azure log.

Reply:
people usually use the `$(uname) == "Darwin"`

Reply:
D'oh, of course. Thanks :)